blaze/compute/sql.py

```diff
@@ -585,6 +592,17 @@ def compute_up(expr, data, **kwargs):
     return table_of_engine(data, expr._name)
 
 
+@dispatch(DateTime, (ClauseElement, sa.sql.elements.ColumnElement))
+def compute_up(expr, data, **kwargs):
+    return getattr(sa.func, expr.attr)(data).label(expr._name)
```

Comment:
It's a bit bittersweet because sqlalchemy itself doesn't abstract away date component access, so this solution will only work if the RDBMs supports access like this

```sql
select day(t.dates) from t
```